lib/proxy/helpers.rb

```diff
@@ -107,4 +107,19 @@ def remote_fqdn(forward_verify=true)
     end
   end
 
+  def ssl_client_cert(request)
+    if request.env.key?('SSL_CLIENT_CERT')
+      request.env['SSL_CLIENT_CERT'].to_s
+    elsif request.env.key?('puma.peercert')
+      request.env['puma.peercert'].to_s
+    else
+      ''
+    end
+  end
+
+  def https?(request)
+    webrick = ['yes', 'on', '1']
+    puma = ['https']
+    (webrick + puma).include? request.env['HTTPS'].to_s
```

Comment:
This should look at the settings and only test for the server in use.

Reply:
For the same reason above I am not doing it. We will be dropping webrick soon and this will become very simple. However I optimized the method a bit, there is no reason allocating and joining arrays each request.